pkg/sql/schemachanger/scbuild/ast_annotator.go

```diff
@@ -29,6 +29,15 @@ func newAstAnnotator(original tree.Statement) (*astAnnotator, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Preserve fields that are not part of the SQL syntax.
+	// ImportRollback is set programmatically and lost during format/parse.
+	if origTruncate, ok := original.(*tree.Truncate); ok {
```

Comment:
I don't love this, but the alternatives are also kind of weird:
1. We use a session variable that can't be set instead of a value on the AST node.
2. We add a `.Clone` method that AST nodes can implement that and use it if it is available.
3. We add some way to construct truncate schema changes that does not require executing a `TRUNCATE` statement.